packages/table/src/table.tsx

```diff
@@ -408,6 +436,29 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         });
     }
 
+    private handleCopy = (e: KeyboardEvent) => {
+        const { grid } = this;
+        const { getCellData, onCopy} = this.props;
+        const { selectedRegions} = this.state;
+
+        if (getCellData == null) {
+            return;
+        }
+
+        // prevent "real" copy from being called
+        e.preventDefault();
+        e.stopPropagation();
+
+        const cells = Regions.enumerateUniqueCells(selectedRegions, grid.numRows, grid.numCols);
+        const sparse = Regions.sparseMapCells(cells, getCellData);
+        if (sparse != null) {
+            const success = Clipboard.copyCells(sparse);
+            if (onCopy != null) {
+                onCopy(success);
```

Comment:
safely invokes a maybe-defined callback. it's precisely this pattern (if function exists then call it) expressed as one type-safe statement.